sdk/next/src/middlewares/proxy.test.ts

```diff
@@ -0,0 +1,126 @@
+    const request = new NextRequest(new URL("https://example.com/proxy/hasura"));
+    const rewriteSpy = spyOn(NextResponse, "rewrite");
+    proxyMiddleware(request);
+    expect(rewriteSpy).toHaveBeenCalledWith(process.env.SETTLEMINT_HASURA_ENDPOINT, {
+      headers: {
+        "x-auth-token": process.env.SETTLEMINT_ACCESS_TOKEN,
+        "x-hasura-admin-secret": process.env.SETTLEMINT_HASURA_ADMIN_SECRET,
+      },
+    });
+  });
+
+  it("should add authentication headers for portal proxy routes", () => {
+    const request = new NextRequest(new URL("https://example.com/proxy/portal"));
+    const rewriteSpy = spyOn(NextResponse, "rewrite");
+    proxyMiddleware(request);
+    expect(rewriteSpy).toHaveBeenCalledWith(process.env.SETTLEMINT_PORTAL_GRAPHQL_ENDPOINT, {
+      headers: {
+        "x-auth-token": process.env.SETTLEMINT_ACCESS_TOKEN,
+      },
+    });
+  });
+
+  it("should add authentication headers for IPFS API proxy routes", () => {
+    const request = new NextRequest(new URL("https://example.com/proxy/ipfs/api"));
+    const rewriteSpy = spyOn(NextResponse, "rewrite");
+    proxyMiddleware(request);
+    expect(rewriteSpy).toHaveBeenCalledWith(process.env.SETTLEMINT_IPFS_API_ENDPOINT, {
+      headers: {
+        "x-auth-token": process.env.SETTLEMINT_ACCESS_TOKEN,
+      },
+    });
+  });
+
+  it("should add authentication headers for IPFS gateway proxy routes", () => {
+    const request = new NextRequest(new URL("https://example.com/proxy/ipfs/gateway"));
+    const rewriteSpy = spyOn(NextResponse, "rewrite");
+    proxyMiddleware(request);
+    expect(rewriteSpy).toHaveBeenCalledWith(process.env.SETTLEMINT_IPFS_GATEWAY_ENDPOINT, {
+      headers: {
+        "x-auth-token": process.env.SETTLEMINT_ACCESS_TOKEN,
+      },
+    });
+  });
+
+  it("should add authentication headers for blockscout proxy routes", () => {
+    const request = new NextRequest(new URL("https://example.com/proxy/blockscout"));
+    const rewriteSpy = spyOn(NextResponse, "rewrite");
+    proxyMiddleware(request);
+    expect(rewriteSpy).toHaveBeenCalledWith(process.env.SETTLEMINT_BLOCKSCOUT_GRAPHQL_ENDPOINT, {
+      headers: {
+        "x-auth-token": process.env.SETTLEMINT_ACCESS_TOKEN,
+      },
+    });
+  });
+
+  it("should handle TheGraph proxy routes with subgraph name", () => {
+    const request = new NextRequest(new URL("https://example.com/proxy/thegraph/graphql/test-subgraph"));
+    const rewriteSpy = spyOn(NextResponse, "rewrite");
+    proxyMiddleware(request);
+    expect(rewriteSpy).toHaveBeenCalledWith(JSON.parse(process.env.SETTLEMINT_THEGRAPH_SUBGRAPHS_ENDPOINTS!)[0], {
+      headers: {
+        "x-auth-token": process.env.SETTLEMINT_ACCESS_TOKEN,
+      },
+    });
+  });
+
+  it("should respect disabled proxy options", () => {
+    const request = new NextRequest(new URL("https://example.com/proxy/hasura"));
+    const nextSpy = spyOn(NextResponse, "next");
```

Comment:
should we also spy on `rewrite` here and check if it is not called?

Reply:
will add this in a next pr